File: packages/shared/src/types/agents.ts
import { TitanEvent } from "./events";

export interface AgentContext {
  taskId: string;
  organizationId: string;
  event: TitanEvent;
  retrievedMemories: Record<string, any>[];
  retrievedDocuments: Record<string, any>[];
}

export type AgentStatus = "SUCCESS" | "FAILED" | "REQUIRES_HUMAN";

export interface AgentOutput {
  agentType: string;
  status: AgentStatus;
  summary: string;
  data: Record<string, any>;
  confidenceScore: number;
}

File: packages/contracts/titan_contracts/agents.py
from pydantic import BaseModel, Field
from typing import Any, Dict, Literal
from .events import TitanEvent

class AgentContext(BaseModel):
    """Context provided to an agent before execution."""
    task_id: str
    organization_id: str
    event: TitanEvent
    retrieved_memories: list[dict] = Field(default_factory=list)
    retrieved_documents: list[dict] = Field(default_factory=list)

class AgentOutput(BaseModel):
    """Standardized output from any agent."""
    agent_type: str
    status: Literal["SUCCESS", "FAILED", "REQUIRES_HUMAN"]
    summary: str
    data: Dict[str, Any] = Field(default_factory=dict)
    confidence_score: float = Field(ge=0.0, le=1.0)
